src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardRetryingRequestResolver.java

```diff
@@ -70,64 +79,53 @@ public int getCurrentReassignmentAttempt()
 
     @Nullable
     @Override
-    public IToken getCurrentlyBeingReassignedRequest()
+    public IToken<?> getCurrentlyBeingReassignedRequest()
     {
         return current;
     }
 
-    public StandardRetryingRequestResolver(final IToken id, final ILocation location)
-    {
-        this.id = id;
-        this.location = location;
-    }
-
     @Override
     public TypeToken<? extends IRetryable> getRequestType()
     {
         return TypeConstants.RETRYABLE;
     }
 
     @Override
-    public boolean canResolve(
-                               @NotNull final IRequestManager manager, final IRequest<? extends IRetryable> requestToCheck)
+    public boolean canResolve(@NotNull final IRequestManager manager, final IRequest<? extends IRetryable> requestToCheck)
     {
         return getCurrentlyBeingReassignedRequest() == null || requestToCheck.getToken() != getCurrentlyBeingReassignedRequest()
                  || getCurrentReassignmentAttempt() < getMaximalTries();
     }
 
     @Nullable
     @Override
-    public List<IToken> attemptResolve(
-                                        @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IRetryable> request)
+    public List<IToken<?>> attemptResolve(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IRetryable> request)
     {
         return ImmutableList.of();
     }
 
-    @Nullable
     @Override
     public void resolve(
-                         @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IRetryable> request) throws RuntimeException
+      @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IRetryable> request) throws RuntimeException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.RuntimeException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)